cmd/ebitenmobile/gomobile.go

```diff
@@ -164,7 +165,8 @@ import (
 	// runtime.Version() is the current executing Go version. For example, this is the version of the toolchain directive in go.mod.
 	// This might differ from the Go command version under the temporary directory.
 	// To avoid the version mismatch, set the toolchain explicitly (#3086).
-	if err := runGo("mod", "edit", "-toolchain="+runtime.Version()); err != nil {
+	toolchainVersion := strings.Split(runtime.Version(), " ")[0]
```

Comment:
```go
toolchainVersion, _, _ := strings.Cut(runtime.Version(), " ")
```

is better.

Reply:
Optional: a more robust way would be

```go
	reVer := regexp.MustCompile(`^go\d+\.\d+\.(\d+)?$`)
	verStr := runtime.Version()
	m := reVer.FindStringSubmatch(verStr)
	if m == nil {
		return tmp, fmt.Errorf("ebitenmobile: unexpected Go version: %s", verStr)
	}
	toolchainVersion := m[0]
```

Reply:
Wouldn’t `go/version.Lang(runtime.Version)` give the value you want?

Reply:
I didn't know that, thanks!

Reply:
https://go.dev/play/p/mJgkLAS_nEw This doesn't work unfortunately.

Reply:
https://go.dev/play/p/44G5Md_xRsZ Also `version.Lang` strips the build version, so this is not ideal.

Reply:
Good feedback, I've pushed a change as suggested.

Reply:
Hmm, this might not work with gotip: https://go.dev/play/p/V9tVDS4adLm?v=gotip

We need to 1. split the string by a space char and 2. check each chunk whether it maches with a regular expression `^go\d+\.\d+(\.\d+)?`